localstack/aws/accounts.py

```diff
@@ -31,30 +31,36 @@ def set_aws_access_key_id(access_key_id: str):
 
 def get_aws_account_id() -> str:
     """Return the AWS account ID for the current context."""
-    return account_id_resolver()
+    try:
+        return REQUEST_CTX_TLS.account_id
+    except AttributeError:
+        LOG.debug(
+            "No Account ID in thread-local storage for thread %s",
+            threading.current_thread().ident,
+        )
+        return DEFAULT_AWS_ACCOUNT_ID
 
 
-def get_default_account_id() -> str:
-    return DEFAULT_AWS_ACCOUNT_ID
-
-
-def get_moto_default_account_id() -> str:
-    return moto.core.DEFAULT_ACCOUNT_ID
-
-
-account_id_resolver = get_default_account_id
-
-#
-# Utils
-#
+def set_aws_account_id(account_id: str) -> None:
+    REQUEST_CTX_TLS.account_id = account_id
 
 
 def get_account_id_from_access_key_id(access_key_id: str) -> str:
     """Return the Account ID associated the Access Key ID."""
-    # This utility ignores IAM mappings.
-    # For now, we assume the client sends Account ID in Access Key ID field.
 
+    # If AWS_ACCESS_KEY_ID has a 12-digit integer value, use it as the account ID
     if re.match(r"\d{12}", access_key_id):
         return access_key_id
-    else:
-        return get_default_account_id()
+
+    elif len(access_key_id) >= 20:
+        # If AWS_ACCESS_KEY_ID has production AWS credentials, ignore them
+        if access_key_id.startswith("ASIA") or access_key_id.startswith("AKIA"):
+            LOG.warning(
+                "Ignoring production AWS credentials provided to LocalStack. Falling back to default account ID."
```

Comment:
@viren-nadkarni Is there any way to silence or change the level of this new warning?

Reply:
@joebowbeer You could set this config var: `PARITY_AWS_ACCESS_KEY_ID=1`

Reply:
Thanks for the suggestion and doc. I wish there were a way to ignore genuine creds `PARITY_AWS_ACCESS_KEY_ID=0` _and_ silence the new warning.